lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
+        matching_sessions.store(session_id, session)
+      end
+    end
+    matching_sessions
+  end
+
+  def filter_sessions_by_search(search_term)
+    matching_sessions = {}
+    
+    framework.sessions.each do |session_id, session|
+      case search_term.split(":")[0]
+      when "last_checkin"
+        if session.respond_to?(:last_checkin) && session.last_checkin && evaluate_search_criteria(session, search_term)
+          matching_sessions.store(session_id, session)
```

Comment:
```suggestion
          matching_sessions[session_id] = session
```
This might just be a personal preference thing but I think this reads better and is much more common in the code base